src/txn/txn_rollback_to_stable.c

```diff
@@ -1432,11 +1432,11 @@ __rollback_to_stable_btree_hs_truncate(WT_SESSION_IMPL *session, uint32_t btree_
 
     /* Open a history store table cursor. */
     WT_ERR(__wt_curhs_open(session, NULL, &hs_cursor));
+    F_SET(hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
 
     /* Walk the history store for the given btree. */
     hs_cursor->set_key(hs_cursor, 1, btree_id);
-    ret = __wt_curhs_search_near_after(session, hs_cursor);
-
+    ret = __wt_curhs_search_near_before(session, hs_cursor);
```

Comment:
I don't clearly understand how this change is fixing the problem that you are facing? Both of these functions should return the key is that exists, if not exists then only it returns the next one.